### PR TITLE
move to environment approach - template service

### PR DESCRIPTION
**Pull request by parimal1987**
Task 8 : move to environment approach - template service

### CODE REVIEW

File: src-appsetting/_MicroserviceTemplate_.WebApi/appsettings-for-using-env.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+   "UseEnvironmentVariables": true
```
Comment (Joseph1977) [inline src-appsetting/_MicroserviceTemplate_.WebApi/appsettings-for-using-env.json]: @parimal1987 why this?  src-msdb, NO need for this use env

File: src-appsetting/_MicroserviceTemplate_.WebApi/appsettings.json
Comment (Joseph1977) [inline src-appsetting/_MicroserviceTemplate_.WebApi/appsettings.json:1]: You must not make ANY change in the old template except the folder name. roll it back!

File: src-env (msdb)/.env/dev-use1/.env [hunk 2]
```diff
@@ -0,0 +1,25 @@
+REGION=USE1 #East USA 1
```
Comment (Joseph1977) [inline src-env (msdb)/.env/dev-use1/.env]: Did you added what we agreed to the readme? i asked you to add the naming convesion of the env:
camel case, sections separate by "_"

File: src-env (msdb)/_MicroserviceTemplate_.EF/DataConfigurations/MyTableDataConfiguration.cs
Comment (Joseph1977) [inline src-env (msdb)/_MicroserviceTemplate_.EF/DataConfigurations/MyTableDataConfiguration.cs:1]: why the values are relevant to setting, this should be a normal table with id (guid) last update time, createtime any any other columns , NOTHING to do with settings/configuration.

File: src-env (msdb)/_MicroserviceTemplate_.EF/DataConfigurations/MyTableDataConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace _MicroserviceTemplate_.EF.DataConfigurations
+{
+    public class MyTableDataConfiguration : IEntityTypeConfiguration<MyTable>
+    {
+        public void Configure(EntityTypeBuilder<MyTable> builder)
+        {
+            builder.HasData(
+                Create("Operation:Add", "Add", "Add a new record."),
```
Comment (Joseph1977) [inline src-env (msdb)/_MicroserviceTemplate_.EF/DataConfigurations/MyTableDataConfiguration.cs]: why? id should be GUID

File: src-env (msdb)/_MicroserviceTemplate_.WebApi/Controllers/MyTableController.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        /// <summary>
+        /// Returns my table entry.
+        /// </summary>
+        /// <param name="id">My table identifier.</param>
+        /// <returns>my table entry.</returns>
+        [HttpGet("{id}")]
+        [Authorize(_MicroserviceTemplate_Policies.MYTABLES_READ)]
+        [ProducesResponseType(typeof(MyTableViewModel), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetByIdAsync([FromRoute] string id)
```
Comment (Joseph1977) [inline src-env (msdb)/_MicroserviceTemplate_.WebApi/Controllers/MyTableController.cs]: ID IS GUID all our tables have id GUID, the only one was settings which the idea was to remove this setting table.

File: src-env (msdb)/_MicroserviceTemplate_.WebApi/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        "Jwt_Authority": "",
+
+        "AccessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJyb2xlIjoidGVzdCJ9.-I0pOAoVASxAkkupXlue8npTy_LBPua_ln_V6Tq_-_8",
+        "BenrazAuthGateway_BaseUrl": "http://localhost:60341",
+
+        "AspNetCoreDbUserName": "__DB_USERNAME__",
+        "AspNetCoreDbPassword": "__DB_PASSWORD__"
+      }
+    },
+    "_MicroserviceTemplate_.WebApi": {
```
Comment (Joseph1977) [inline src-env (msdb)/_MicroserviceTemplate_.WebApi/Properties/launchSettings.json:48]: why we have 2 env in the launch ?

File: src-env (msdb)/_MicroserviceTemplate_.WebApi/Startup.cs [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+
+                    options.Audience = Configuration["Jwt:Audience"];
+                    options.Authority = Configuration["Jwt:Authority"];
+
+                });
+
+            services
+                .AddAuthorization(options =>
+                {
+                    options.AddClaimsPolicy(
```
Comment (Joseph1977) [inline src-env (msdb)/_MicroserviceTemplate_.WebApi/Startup.cs:210]: you need to add new claim file in the doc folder for this template

**Review by Joseph1977 — CHANGES_REQUESTED** [review]
added comments.

File: src-appsetting/_MicroserviceTemplate_.WebApi/appsettings-for-using-env.json (hunk 1 above)
Comment (parimal1987) [reply src-appsetting/_MicroserviceTemplate_.WebApi/appsettings-for-using-env.json]: @Joseph1977 In the initial commit you have added this file in the old template and I used the same branch for the development.
Let me remove it.

File: src-appsetting/_MicroserviceTemplate_.WebApi/appsettings.json
Comment (parimal1987) [reply src-appsetting/_MicroserviceTemplate_.WebApi/appsettings.json:1]: @Joseph1977 In the initial commit you have added this file in the old template and I used the same branch for the development.
Let me roll back it.

File: src-env (msdb)/.env/dev-use1/.env (hunk 2 above)
Comment (parimal1987) [reply src-env (msdb)/.env/dev-use1/.env]: Yes I added

File: src-env (msdb)/_MicroserviceTemplate_.EF/DataConfigurations/MyTableDataConfiguration.cs
Comment (parimal1987) [reply src-env (msdb)/_MicroserviceTemplate_.EF/DataConfigurations/MyTableDataConfiguration.cs:1]: Changed id column type to guid from string.

File: src-env (msdb)/_MicroserviceTemplate_.WebApi/Startup.cs (hunk 6 above)
Comment (parimal1987) [reply src-env (msdb)/_MicroserviceTemplate_.WebApi/Startup.cs:210]: Added new claim file with the below name
1. Microservice template claims env (msdb).sql
2. Microservice template claims env (msdb).xlsx

File: src-env (msdb)/_MicroserviceTemplate_.WebApi/Properties/launchSettings.json (hunk 5 above)
Comment (parimal1987) [reply src-env (msdb)/_MicroserviceTemplate_.WebApi/Properties/launchSettings.json:48]: I removed environment variables from the **_MicroserviceTemplate_.WebApi** profile.
While local development we used **IIS Express** profile.

File: src-env (msdb)/.env/dev-use1/.env [hunk 7]
```diff
@@ -1,23 +1,30 @@
...
-SERVICENAME=_microservicetemplate_-server
-SERVICE=MSTS
+Region=USE1 #East USA 1
+Environment=DEV
+ServiceName=_microservicetemplate_-server
+Service=MSTS
 
 AllowedHosts=*
 DisableSwagger=false
+InjectDBCredentialFromEnvironment=false -------- obviouse!!!
```
Comment (Joseph1977) [inline src-env (msdb)/.env/dev-use1/.env]: rempve the " -------- obviouse!!!" 
it is false or true, and default should be true, move it above ASPNETCORE_DB_USERNAME

File: src-env (msdb)/.env/dev-usw1/.env [hunk 8]
```diff
@@ -1,25 +1,32 @@
...
-SERVICENAME=_microservicetemplate_-server
-SERVICE=MSTS
+Region=USW1 #West USA 1
+Environment=DEV
+ServiceName=_microservicetemplate_-server
+Service=MSTS
 
 AllowedHosts=*
 DisableSwagger=false
+InjectDBCredentialFromEnvironment=false -------- obviouse!!!
```
Comment (Joseph1977) [inline src-env (msdb)/.env/dev-usw1/.env:8]: same

File: src-env (msdb)/.env/prod-usc1/.env [hunk 9]
```diff
@@ -0,0 +1,32 @@
+Region=USC1 #Center USA 1
+Environment=PROD
+ServiceName=_microservicetemplate_-server
+Service=MSTS
+
+AllowedHosts=*
+DisableSwagger=false
+InjectDBCredentialFromEnvironment=false -------- obviouse!!!
```
Comment (Joseph1977) [inline src-env (msdb)/.env/prod-usc1/.env:8]: same

File: src-env (msdb)/.env/qa-use1/.env [hunk 10]
```diff
@@ -0,0 +1,32 @@
+Region=USE1 #East USA 1
+Environment=QA
+ServiceName=_microservicetemplate_-server
+Service=MSTS
+
+AllowedHosts=*
+DisableSwagger=false
+InjectDBCredentialFromEnvironment=false -------- obviouse!!!
```
Comment (Joseph1977) [inline src-env (msdb)/.env/qa-use1/.env:8]: same

File: src-env (msdb)/.env/sb-euw1/.env [hunk 11]
```diff
@@ -0,0 +1,32 @@
+Region=USE1  #East USA 1
+Environment=SB  #Sandbox
+ServiceName=_microservicetemplate_-server
+Service=MSTS
+
+AllowedHosts=*
+DisableSwagger=false
+InjectDBCredentialFromEnvironment=false -------- obviouse!!!
```
Comment (Joseph1977) [inline src-env (msdb)/.env/sb-euw1/.env:8]: same

File: src-env (msdb)/_MicroserviceTemplate_.WebApi/Models/MyTables/ChangeMyTableViewModel.cs [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+{
+    /// <summary>
+    /// Change my table view model.
+    /// </summary>
+    public class ChangeMyTableViewModel
+    {
+        /// <summary>
+        /// Value.
+        /// </summary>
+        public string Value { get; set; }
```
Comment (Joseph1977) [inline src-env (msdb)/_MicroserviceTemplate_.WebApi/Models/MyTables/ChangeMyTableViewModel.cs:11]: description missing, not big deal

**Review by Joseph1977 — APPROVED** [review]
@parimal1987  few small comments, pls resolve and complete the merge

**Pull request merged**